tmt/steps/report/polarion.py

```diff
@@ -73,16 +73,14 @@ def go(self, *args: Any, **kwargs: Any) -> None:
             '*property[@name="polarion-project-span-ids"]')
 
         for result in self.step.plan.execute.results():
-            if not result.id:
+            if not result.ids or not any(result.ids.values()):
                 raise tmt.utils.ReportError(
                     f"Test Case {result.name} is not exported to Polarion, "
                     "please run 'tmt tests export --how polarion' on it")
-            work_item_id, test_project_id = get_polarion_ids(
-                PolarionWorkItem.query(
-                    result.id, fields=['work_item_id', 'project_id']))
+            work_item_id, test_project_id = find_polarion_case_ids(result.ids)
 
             assert work_item_id is not None
-            assert test_project_id is not None
+            assert test_project_id is not None, "Test case missing or not found in Polarion"
```

Comment:
As asserts should not be used for "production" checks, shall we rather raise an exception or print a warning instead?

Reply:
This assert is part of the typing changes, I only added a useful message so we don't have to have another, basically the same, check.

Reply:
So... it's ok that the message will never be shown to the user and the code will just go on? Why adding it here then?

Reply:
It will be shown, when a test case is not found in Polarion then this message will appear as the reason for the assertion error, instead of just a plain assertion error.

Reply:
If this assert can fail depending on user input (which looks it can) then it should be captured properly and whole 'go()' should raise with 'tmt.utils.ReportError`. 'AssertionError' isn't something tmt should show the user.

Assert use for typing is just to make it clear for mypy that while some function might return None, it is not a valid value past the assert.

Reply:
Well sure, but catching the error properly would look like
```
if test_project_id is None:
    raise ...
assert...
```
which would make both the asserts just a `nop` taking time to check for something that was already checked

Reply:
I believe you don't need no assert after that - mypy knows that 'None' branch is already processed.